src/test/java/com/edmunds/tools/databricks/maven/ImportWorkspaceMojoTest.java

```diff
@@ -56,18 +65,37 @@ public void testImportWorksWithProperPath() throws Exception {
     // Cannot use expected exceptions with a MojoExecutionException...
     @Test
     public void testImportFailsWithImproperPath() throws Exception {
+
+        ImportWorkspaceMojo underTest = getNoOverridesMojo(GOAL, "_fails_validation");
+
         File workspacePath = new File(this.getClass().getResource("/notebooks")
-            .getPath());
+                .getPath());
         underTest.validate = true;
         underTest.setSourceWorkspacePath(workspacePath);
-        Mockito.when(project.getGroupId()).thenReturn("com.edmunds.nottest");
-        Mockito.when(project.getArtifactId()).thenReturn("mycoolartifact");
-        underTest.setProject(project);
         try {
             underTest.execute();
         } catch (MojoExecutionException e) {
+            assertThat(e.getMessage(), containsString("JOB NAME VALIDATION FAILED [ILLEGAL VALUE]:\n" +
+                    "Expected: [failed-test] but found: [test]"));
             return;
         }
         fail();
     }
+
+    @Test
+    public void testImportWorksWithProperPathAndOverrides() throws Exception {
+        ImportWorkspaceMojo underTest = getOverridesMojo(GOAL);
+
+        File workspacePath = new File(this.getClass().getResource("/notebooks")
+                .getPath());
+
+        underTest.validate = true;
+        underTest.setSourceWorkspacePath(workspacePath);
+
+        Mockito.verify(libraryService, times(0)).install(Matchers.anyString(), Matchers.any());
+
+        underTest.execute();
+
+        Mockito.verify(workspaceService, Mockito.times(3)).importWorkspace(Mockito.any(ImportWorkspaceRequest.class));
```

Comment:
could be valuable to test against specific ImportWorkspaceRequests

Reply:
perhaps. I think maybe that is adding something to the request though. Perhaps it would be best to add a story for this? e.g. "Add additional tests to ImportWorkspace functionality"?

Reply:
Nope, I track you now. I'll update the test.

Reply:
oh i had no idea these were existing tests. My bad. Your call